File: output/xCodeEval_8000/C++/Python/91520aa153f53578d11861d4a117e02f.py
import math

def to_int(s):
    x = 0
    for i in range(len(s)):
        x *= 10
        x += int(s[i])
    return x

def solve():
    s = input()
    st10 = [1] * 20
    for i in range(1, 20):
        st10[i] = st10[i-1] * 10
    s = s[::-1]
    while s[-1] < '0' or s[-1] > '9':
        s = s[:-1]
    s = s[::-1]
    VEC = []
    for i in range(len(s), 0, -1):
        VEC.append(to_int(s))
        s = s[::-1]
        s = s[:-1]
        s = s[::-1]
    VEC = VEC[::-1]
    SET = set()
    ans = 0
    for i in range(len(VEC)):
        x = VEC[i]
        while x < 1989 or x in SET:
            x += st10[i+1]
        SET.add(x)
        ans = x
    print(ans)

t = int(input())
for _ in range(t):
    solve()

File: output/xCodeEval_8000/C++/C/91520aa153f53578d11861d4a117e02f.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define maxN 100300

int to_int(char* s) {
    int x = 0;
    for (int i = 0; i < strlen(s); ++i) {
        x *= 10;
        x += (s[i] - '0');
    }
    return x;
}

void solve() {
    char s[100];
    scanf("%s", s);
    int st10[20];
    st10[0] = 1;
    for (int i = 1; i < 20; ++i) st10[i] = st10[i - 1] * 10;
    strrev(s);
    while (s[strlen(s) - 1] < '0' || s[strlen(s) - 1] > '9') s[strlen(s) - 1] = '\0';
    strrev(s);
    int VEC[100];
    int size = 0;
    for (int i = strlen(s); i > 0; --i) {
        VEC[size++] = to_int(s);
        strrev(s);
        s[strlen(s) - 1] = '\0';
        strrev(s);
    }
    for (int i = size - 1; i >= 0; --i) {
        int x = VEC[i];
        while (x < 1989) {
            int isFound = 0;
            for (int j = 0; j < size; ++j) {
                if (VEC[j] == x) {
                    isFound = 1;
                    break;
                }
            }
            if (isFound) {
                x += st10[i + 1];
            } else {
                break;
            }
        }
        VEC[i] = x;
    }
    int SET[100];
    int setSize = 0;
    int ans = 0;
    for (int i = 0; i < size; ++i) {
        int x = VEC[i];
        int isFound = 0;
        for (int j = 0; j < setSize; ++j) {
            if (SET[j] == x) {
                isFound = 1;
                break;
            }
        }
        if (!isFound) {
            SET[setSize++] = x;
            ans = x;
        }
    }
    printf("%d\n", ans);
}

int main(void) {
    int t;
    scanf("%d", &t);
    while (t--) {
        solve();
    }
    return 0;
}
